### ISSUE TITLE
Steam District Heating - Networks

### ISSUE THREAD

**Issue by khinkelman** [lbl-srg/modelica-buildings#2904]
@bravache Could you please create a new branch for this issue at your earliest convenience? These new models are **not** dependent on any existing open branches. 

This issue is to add the steam district heating models to the `DHC.Networks.Steam` package, as previously agreed upon (green highlighted area in the image below). This work is part of the URBANopt project, and the merging is **time sensitive** in order to submit the paper "Towards Open-Source Modelica Models for Steam-Based District Heating Systems" to the 1st International workshop on "Open Source Modelling and Simulation of Energy Systems", with a final submission **deadline of March 10, 2022**. 

![image](https://user-images.githubusercontent.com/44682485/156035803-c549f05b-30a2-46b1-a7b6-58b56a0d72cd.png)

**Model Descriptions**
These network models follow the same structure as the existing `DHC.Networks` models developed by @AntoineGautier. However, they have been customized for steam systems that use the split-medium approach (as described in [(Preprint) Hinkelman et al. 2022, Energy.](https://www.researchgate.net/publication/358860784_A_Fast_and_Accurate_Modeling_Approach_for_Water_and_Steam_Thermodynamics_with_Practical_Applications_in_District_Heating_System_Simulation)). The base classes allow users to replace the pipe models in `Connection2Pipe` and `Distribution2Pipe` with alternates while retaining compatible interfaces among all models in the `DHC.Networks` package.

If you have any further questions, please let me know. Many thanks!

@SaranyaAnbarasu: cc'ing you so you are aware and can track.

**Comment by JayHuLBL** [comment]
@khinkelman @bravache I just created the new branch [issue2904_steamnetwork](https://github.com/lbl-srg/modelica-buildings/tree/issue2904_steamnetwork) for the development.

**Comment by khinkelman** [comment]
Many thanks!

**Comment by bravache** [comment]
@JayHuLBL @khinkelman Sorry for the delay, thanks Jianjun for taking care of that!

**Comment by khinkelman** [comment]
The current steam network models include lossless pipes on the steam supply side. Thus, the steam distribution network model example is structurally singular for open-loop examples with mass flow and/or pressure states declared at the network boundaries. 
To keep with the original example model implemented, the `BuildingTimeSeries` model and its `SteamTrap` base class will be added to this PR. The green shaded region above has been adjusted accordingly.

**Comment by bravache** [comment]
Well, this has already been merged to `lbl-srg`. I might create another branch to pull everything bit by bit though. This way we can have a PR for the changes that are specific to this, and also merge everything to master together. Does that make sense?

**Comment by khinkelman** [comment]
Yes, that makes sense. I'll open the PR for this issue now so that we can see the changes and keep moving everything forward bit by bit. Thanks again for your help!
